producer-kafka/src/test/java/io/stargate/producer/kafka/KafkaProducerActivatorTest.java

```diff
@@ -104,4 +157,126 @@ public void shouldRegisterServiceOnlyOnceEvenIfStartIsCalledSecondTime()
         .registerService(eq(CDCProducer.class), any(CDCProducer.class), eq(null));
     assertThat(kafkaProducerActivator.started).isTrue();
   }
+
+  @Test
+  public void shouldNotStartIfOnlyMetricsServiceIsRegistered() throws InvalidSyntaxException {
+    // given
+    BundleContext bundleContext = mock(BundleContext.class);
+    KafkaProducerActivator kafkaProducerActivator = new KafkaProducerActivator();
+    kafkaProducerActivator.start(bundleContext);
+    ServiceEvent metricsServiceEvent = mock(ServiceEvent.class);
+    try (MockedStatic<BundleUtils> bundleUtilsMock = mockStatic(BundleUtils.class)) {
+      mockMetricsServiceNotification(bundleContext, metricsServiceEvent, bundleUtilsMock);
```

Comment:
If we're starting to use `BundleUtils` as a sort of "context" object for interacting with the OSGi env., wouldn't it be nice to refactor it to be an interface perhaps, and just have a regular mock in tests? Mocking static methods does not look elegant from my POV... just my 2 cents.

Reply:
I am also not a fan of this static method. I would be happy to refactor it, but for sure, in a separate PR (issue).
I think that we have a task with refactoring OSGi Activators' logic, but I cannot find it.

Reply:
I think we talked about OSGi activation, but never files an issue :) +1 to opening one now.

Reply:
Created: https://github.com/stargate/stargate/issues/364